alibi_detect/od/processor/base.py

```diff
@@ -0,0 +1,13 @@
+import numpy as np
+
+
+class BaseProcessor():
+    def __init__(self, detectors):
+        self.detectors = detectors
+
+    def __call__(self, X):
+        B, _ = X.shape
```

Comment:
This implies that the input `x` has shape `(n, d)`, but would fail for e.g. `x = (n, d1, d2)`.